iml-graphql-queries/src/log.rs

```diff
@@ -0,0 +1,123 @@
+                        facility
+                        fqdn
+                        message
+                        message_class: messageClass
+                        severity
+                        tag
+                    }
+                    meta {
+                        total_count: totalCount
+                    }
+                }
+            }
+        "#;
+
+    #[derive(Debug, serde::Serialize)]
+    pub struct Vars {
+        limit: Option<usize>,
+        offset: Option<usize>,
+        dir: Option<SortDir>,
+        message: Option<String>,
+        fqdn: Option<String>,
+        tag: Option<String>,
+        start_datetime: Option<String>,
+        end_datetime: Option<String>,
+        message_class: Option<Vec<MessageClass>>,
+        severity: Option<LogSeverity>,
+    }
+
+    #[derive(Debug)]
+    pub struct Builder {
+        vars: Vars,
+    }
+
+    impl Builder {
+        pub fn new() -> Self {
+            Self {
+                vars: Vars {
+                    limit: None,
+                    offset: None,
+                    dir: None,
+                    message: None,
+                    fqdn: None,
+                    tag: None,
+                    start_datetime: None,
+                    end_datetime: None,
+                    message_class: None,
+                    severity: None,
+                },
+            }
+        }
+
+        pub fn with_limit(mut self, limit: usize) -> Self {
+            self.vars.limit = Some(limit);
+            self
+        }
+
+        pub fn with_offset(mut self, offset: usize) -> Self {
+            self.vars.offset = Some(offset);
+            self
+        }
+
+        pub fn with_dir(mut self, dir: SortDir) -> Self {
+            self.vars.dir = Some(dir);
+            self
+        }
+
+        pub fn with_message(mut self, message: impl ToString) -> Self {
+            self.vars.message = Some(message.to_string());
+            self
+        }
+
+        pub fn with_fqdn(mut self, fqdn: impl ToString) -> Self {
+            self.vars.fqdn = Some(fqdn.to_string());
+            self
+        }
+
+        pub fn with_tag(mut self, tag: impl ToString) -> Self {
+            self.vars.tag = Some(tag.to_string());
+            self
+        }
+
+        pub fn with_start_datetime(mut self, start_datetime: impl ToString) -> Self {
+            self.vars.start_datetime = Some(start_datetime.to_string());
+            self
+        }
+
+        pub fn with_end_datetime(mut self, end_datetime: impl ToString) -> Self {
+            self.vars.end_datetime = Some(end_datetime.to_string());
+            self
+        }
+
+        pub fn with_message_class(mut self, message_class: &[MessageClass]) -> Self {
+            self.vars.message_class = Some(message_class.to_vec());
+            self
+        }
+
+        pub fn with_severity(mut self, severity: &LogSeverity) -> Self {
+            self.vars.severity = Some(severity.clone());
```

Comment:
Looks like there are a couple of clippy warnings here.

Reply:
Done

Reply:
Still warnings here:

![Screen Shot 2020-10-26 at 10 24 50 AM](https://user-images.githubusercontent.com/458717/97184633-81217c80-1775-11eb-9104-839f1be3b12d.png)